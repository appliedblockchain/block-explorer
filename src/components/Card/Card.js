import React from 'react'
import PropTypes from 'prop-types'
import styles from './style.module.css'

const Card = ({ children, headerText, width, height }) =>
  <div className={styles.cardContainer} style={{ width, height }}>
    {
      headerText &&
      <div className={styles.cardHeader}>
        {headerText}
      </div>
    }
    {children}
  </div>

Card.defaultProps = {
  width: 400
}

Card.propTypes = {
  children: PropTypes.node,
  headerText: PropTypes.string,
  width: PropTypes.oneOfType([ PropTypes.number, PropTypes.string ]),
  height: PropTypes.number
}

export default Card
