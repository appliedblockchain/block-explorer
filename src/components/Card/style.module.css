.cardContainer {
  background-color: white;
  border: 1px solid lightgrey;
  border-radius: 3px;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
  margin-top: 20px;
  overflow: hidden;
  margin: 20px;
}

.cardHeader {
  font-size: 18px;
  padding: 10px;
  width: 100%;
  border-bottom: 1px solid #C2DEEA;
  box-sizing: border-box;
}