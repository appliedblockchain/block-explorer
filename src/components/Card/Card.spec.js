import React from 'react'
import renderer from 'react-test-renderer'
import Card from './Card'

it('Card renders correctly', () => {
  const tree = renderer
    .create(
      <Card headerText="Card Header" width={500}>
        <div>Card content</div>
      </Card>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
