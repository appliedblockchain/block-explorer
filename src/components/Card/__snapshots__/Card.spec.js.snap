// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="cardContainer"
  style={
    Object {
      "height": undefined,
      "width": 500,
    }
  }
>
  <div
    className="cardHeader"
  >
    Card Header
  </div>
  <div>
    Card content
  </div>
</div>
`;
